docs/user_guide/building_blocks/config_objects.md

```diff
@@ -53,9 +121,340 @@ given via the ```--variables``` argument, accessing information from columns bas
+An example for a CMS luminosity of 2017 with uncertainty sources given as relative errors is given below.
+
+```python
+from scinum import Number
+cfg.x.luminosity = Number(41480, {
+    "lumi_13TeV_2017": 0.02j,
+    "lumi_13TeV_1718": 0.006j,
+    "lumi_13TeV_correlated": 0.009j,
+})
+```
+
+
+### Defaults
+
+Default values can be given for several command line parameters in columnflow, using the `cfg.x.default_{parameter}` entry in the Config object.
+The expected format is a single string containing the name of the object to be used as default for parameters accepting only one argument and a tuple for parameters accepting several arguments.
+The command-line arguments supporting a default value from the Config object are given in the cms_minimal example of the analysis_templates and shown again below:
+
+```python
+cfg.x.default_calibrator = "example"
+cfg.x.default_selector = "example"
+cfg.x.default_producer = "example"
+cfg.x.default_ml_model = None
+cfg.x.default_inference_model = "example"
+cfg.x.default_categories = ("incl",)
+cfg.x.default_variables = ("n_jet", "jet1_pt")
+cfg.x.default_weight_producer = "example"
+```
+
+
+### Groups
+
+It is also possible to create groups, which allow to conveniently loop over certain command-line parameters.
+This is done with the `cfg.x.{parameter}_group` arguments.
+The expected format of the group is a dictionary containing the custom name of the groups as keys and the list of the parameter values as values.
+The name of the group can then be given as command-line argument instead of the single values.
+An example with a selector_steps group is given below.
+
+```python
+# selector step groups for conveniently looping over certain steps
+# (used in cutflow tasks)
+cfg.x.selector_step_groups = {
+    "default": ["muon", "jet"],
+}
+```
+
+With this group defined in the Config object, running over the "muon" and "jet" selector_steps in this order in a cutflow task can done with the argument `--selector-steps default`.
+
+All other parameters for which groups are possible are given below:
+```python
+# process groups for conveniently looping over certain processs
+# (used in wrapper_factory and during plotting)
+cfg.x.process_groups = {}
+
+# dataset groups for conveniently looping over certain datasets
+# (used in wrapper_factory and during plotting)
+cfg.x.dataset_groups = {}
+
+# category groups for conveniently looping over certain categories
+# (used during plotting)
+cfg.x.category_groups = {}
+
+# variable groups for conveniently looping over certain variables
+# (used during plotting)
+cfg.x.variable_groups = {}
+
+# shift groups for conveniently looping over certain shifts
+# (used during plotting)
+cfg.x.shift_groups = {}
+```
```

Comment:
There is also the `selector_step_groups`, `calibrator_groups`,  `producer_groups`, `ml_model_groups`

Reply:
That parameter is also rather new, the group resolving is done here: https://github.com/columnflow/columnflow/blob/master/columnflow/tasks/framework/plotting.py#L166